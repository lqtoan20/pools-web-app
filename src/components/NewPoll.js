import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../actions/questions";
import UIkit from "uikit";

const NewPoll = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [firstOption, setFirstOption] = useState("");
  const [secondOption, setSecondOption] = useState("");

  const handleFirstOptionChange = (e) => {
    const value = e.target.value;
    setFirstOption(value);
  };

  const handleSecondOptionChange = (e) => {
    const value = e.target.value;
    setSecondOption(value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(handleAddQuestion(firstOption, secondOption));
    UIkit.notification("Question added successfully!", {
      status: "success",
    });
    setFirstOption("");
    setSecondOption("");
    navigate("/");
  };

  return (
    <div>
      <h1 className="uk-text-3x1 uk-text-bold uk-margin-medium-top">
        New Poll
      </h1>
      <form onSubmit={handleSubmit} className="uk-form-stacked">
        <div className="uk-margin">
          <label htmlFor="firstOption" className="uk-form-label">
            First Option
          </label>
          <div className="uk-form-controls">
            <input
              value={firstOption}
              onChange={handleFirstOptionChange}
              type="text"
              name="firstOption"
              id="firstOption"
              className="uk-input"
              data-testid="first-option"
            />
          </div>
        </div>

        <div className="uk-margin">
          <label htmlFor="secondOption" className="uk-form-label">
            Second Option
          </label>
          <div className="uk-form-controls">
            <input
              value={secondOption}
              onChange={handleSecondOptionChange}
              type="text"
              name="secondOption"
              id="secondOption"
              data-testid="second-option"
              className="uk-input"
            />
          </div>
        </div>

        <div className="uk-margin uk-flex uk-flex-right">
          <button
            type="submit"
            data-testid="submit-poll"
            className="uk-button uk-button-primary"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default NewPoll;
