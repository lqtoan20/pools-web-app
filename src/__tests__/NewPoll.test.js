import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../stores/store";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import NewPoll from "../components/NewPoll";

describe("NewPoll", () => {
  it("Should render component and display all elements", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();

    const firstOptionInput = component.getByTestId("first-option");
    const secondOptionInput = component.getByTestId("second-option");

    expect(firstOptionInput).toBeInTheDocument();
    expect(secondOptionInput).toBeInTheDocument();

    fireEvent.change(firstOptionInput, { target: { value: "Test 1" } });
    fireEvent.change(secondOptionInput, { target: { value: "Test 2" } });
    expect(firstOptionInput.value).toBe("Test 1");
    expect(secondOptionInput.value).toBe("Test 2");
  });
});
