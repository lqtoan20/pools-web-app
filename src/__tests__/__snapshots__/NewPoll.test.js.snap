// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll Should render component and display all elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="uk-text-3x1 uk-text-bold uk-margin-medium-top"
        >
          New Poll
        </h1>
        <form
          class="uk-form-stacked"
        >
          <div
            class="uk-margin"
          >
            <label
              class="uk-form-label"
              for="firstOption"
            >
              First Option
            </label>
            <div
              class="uk-form-controls"
            >
              <input
                class="uk-input"
                data-testid="first-option"
                id="firstOption"
                name="firstOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="uk-margin"
          >
            <label
              class="uk-form-label"
              for="secondOption"
            >
              Second Option
            </label>
            <div
              class="uk-form-controls"
            >
              <input
                class="uk-input"
                data-testid="second-option"
                id="secondOption"
                name="secondOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="uk-margin uk-flex uk-flex-right"
          >
            <button
              class="uk-button uk-button-primary"
              data-testid="submit-poll"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="uk-text-3x1 uk-text-bold uk-margin-medium-top"
      >
        New Poll
      </h1>
      <form
        class="uk-form-stacked"
      >
        <div
          class="uk-margin"
        >
          <label
            class="uk-form-label"
            for="firstOption"
          >
            First Option
          </label>
          <div
            class="uk-form-controls"
          >
            <input
              class="uk-input"
              data-testid="first-option"
              id="firstOption"
              name="firstOption"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="uk-margin"
        >
          <label
            class="uk-form-label"
            for="secondOption"
          >
            Second Option
          </label>
          <div
            class="uk-form-controls"
          >
            <input
              class="uk-input"
              data-testid="second-option"
              id="secondOption"
              name="secondOption"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="uk-margin uk-flex uk-flex-right"
        >
          <button
            class="uk-button uk-button-primary"
            data-testid="submit-poll"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
